client/components/UserList/CollaboratorMultiSelect/CollaboratorMultiSelect.tsx

```diff
@@ -0,0 +1,157 @@
+import { useAssessmentSections } from '@client/store/assessment'
+import { UserManagementActions } from '@client/store/userManagement'
+
+type Option = {
+  value: string
+  label: string
+}
+
+type Props = {
+  user: User
+  disabled?: boolean
+}
+
+const CollaboratorMultiSelect: React.FC<Props> = ({ user, disabled }) => {
+  const dispatch = useAppDispatch()
+  const { i18n } = useTranslation()
+
+  const ref = useRef(null)
+  const firstUpdate = useRef(true)
+
+  const [open, setOpen] = useState<boolean>(false)
+
+  const assessmentSections = useAssessmentSections()
+
+  const optionAll = { value: 'all', label: 'all' }
+  const optionNone = { value: 'none', label: 'none' }
+  const options = [
+    optionAll,
+    optionNone,
+    ...assessmentSections
+      .reduce((prev, curr) => [...prev, ...curr.subSections], [])
+      .filter((subSection: SubSection) => subSection.props.anchor)
+      .map((subSection: SubSection): { value: string; label: string } => {
+        return { value: subSection.uuid, label: subSection.props.anchor }
+      }),
+  ]
+  const properties = (user.roles[0].props as unknown as CollaboratorProps) || undefined
+  const sections = Objects.isEmpty(properties) ? 'none' : properties.sections
+  const selectedValues: Array<string> = typeof sections === 'string' ? [sections] : Object.keys(sections)
+
+  const [selectedOptions, setSelectedOptions] = useState<Array<Option>>(
+    selectedValues.map((selectedValue) => {
+      const option = options.find((opt) => opt.value === selectedValue)
+      return {
+        value: selectedValue,
+        label: option ? option.label : 'ooo',
+      }
+    })
+  )
+
+  const localizeOption = (option: Option) => {
+    const { value, label } = option
+    return ['all', 'none'].includes(value) ? i18n.t(`contactPersons.${label}`) : label
+  }
+
+  const removeOption = (option: Option): void => {
+    const { value } = option
+    if (value === 'all') setSelectedOptions([optionNone])
+    else if (value === 'none') setSelectedOptions([optionAll])
+    else setSelectedOptions(selectedOptions.filter((opt) => opt.value !== value))
+  }
+
+  const addOption = (option: Option): void => {
+    const { value } = option
+    if (value === 'all') setSelectedOptions([optionAll])
+    else if (value === 'none') setSelectedOptions([optionNone])
+    else setSelectedOptions([...selectedOptions.filter((opt) => !['all', 'none'].includes(opt.value)), option])
+  }
+
+  const toggleOption = (option: Option): void =>
+    selectedOptions.find((opt) => opt.value === option.value) ? removeOption(option) : addOption(option)
+
+  const handleClickOutside = (e: MouseEvent) =>
+    ref?.current && ref.current && !ref.current.contains(e.target) && setOpen(false)
+
+  useEffect(() => {
+    document.addEventListener('click', handleClickOutside)
+
+    return () => {
+      document.removeEventListener('click', handleClickOutside)
+    }
+  }, [])
+
+  useEffect(() => {
```

Comment:
use the useOnUpdate hook

Reply:
wow, I didn't know about this hook. It does exactly what I implemented following an example found on the Internet.